backend/Origam.DA.Service/AbstractSqlDataService.cs

```diff
@@ -478,174 +503,186 @@ internal override IDbTransaction GetTransaction(string transactionId, IsolationL
+									entity: entity,
+									actionToExecute: () =>
 									{
-										profiler.ExecuteAndLogStoreActionDuration(
-											entity: entity,
-											actionToExecute: () =>
-										{
-											ExecuteUpdate(query, transactionId, profile,
-												ds, transaction, connection,
-												deletedRowIds, 
-	                                            changedTable,
-												rowState, entity, rowCount, 
-	                                            forceBulkInsert);
-										});
-									}
-								}
-
-								// finally add the table to a list of changed tables so 
-								// changes can be accepted later
-								if (!changedTables.Contains(changedTable.TableName))
-								{
-									changedTables.Add(changedTable.TableName);
-								}
+										ExecuteUpdate(query, transactionId, 
+											profile, dataStructure, 
+											transaction, connection,
+											deletedRowIds, changedTable,
+											rowState, entity, rowCount, 
+											forceBulkInsert);
+									});
 							}
-						}						
+						}
+						// finally add the table to a list of changed tables so 
+						// changes can be accepted later
+						if(!changedTables.Contains(changedTable.TableName))
+						{
+							changedTables.Add(changedTable.TableName);
+						}
 					}
 				}
 				// execute the state machine events
-				if(stateMachine != null && query.FireStateMachineEvents)
+				if((stateMachine != null) && query.FireStateMachineEvents)
 				{
-					stateMachine.OnDataChanged(dataset, changedTables, transactionId);
+					stateMachine.OnDataChanged(
+						dataset, changedTables, transactionId);
 				}
 				// accept changes
-				this.AcceptChanges(dataset, changedTables, query, transactionId, userProfile);
+				AcceptChanges(dataset, changedTables, query, transactionId, 
+					userProfile);
 				profiler.LogRememberedExecutionTimes();
                 // delete attachments if any
-                if (deletedRowIds.Count > 0 && query.SynchronizeAttachmentsOnDelete)
+                if((deletedRowIds.Count > 0) 
+                && query.SynchronizeAttachmentsOnDelete)
                 {
-                    IAttachmentService attachmentService = ServiceManager.Services.GetService(typeof(IAttachmentService)) as IAttachmentService;
-                    foreach (Guid recordId in deletedRowIds)
+                    var attachmentService = ServiceManager.Services
+	                    .GetService<IAttachmentService>();
+                    foreach(Guid recordId in deletedRowIds)
                     {
-                        attachmentService.RemoveAttachment(recordId, transactionId);
+                        attachmentService.RemoveAttachment(
+	                        recordId, transactionId);
                     }
                 }
 				if(newTransaction)
 				{
 					ResourceMonitor.Commit(transactionId);
 				}
 			}
-			catch(DBConcurrencyException e)
+			catch(DBConcurrencyException ex)
```

Comment:
Before you changed variable name "ex" to "exception", now you change "e" to "ex"? I think that "ex" is just fine as name of a chatch variable, just like "i" is ok in a for loop variable. I suggest we use "ex" every where.

Reply:
The applied rule is tu use `ex` in `catch` and to use `exception` as argument of the function.

Reply:
ok